core/src/test/java/com/datastrato/gravitino/listener/api/event/TestCatalogEvent.java

```diff
@@ -0,0 +1,248 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.api.event;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.CatalogChange;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.catalog.CatalogDispatcher;
+import com.datastrato.gravitino.catalog.CatalogEventDispatcher;
+import com.datastrato.gravitino.exceptions.GravitinoRuntimeException;
+import com.datastrato.gravitino.listener.DummyEventListener;
+import com.datastrato.gravitino.listener.EventBus;
+import com.datastrato.gravitino.listener.api.info.CatalogInfo;
+import com.google.common.collect.ImmutableMap;
+import java.util.Arrays;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class TestCatalogEvent {
+  private CatalogEventDispatcher dispatcher;
+  private CatalogEventDispatcher failureDispatcher;
+  private DummyEventListener dummyEventListener;
+  private Catalog catalog;
+
+  @BeforeAll
+  void init() {
+    this.catalog = mockCatalog();
+    this.dummyEventListener = new DummyEventListener();
+    EventBus eventBus = new EventBus(Arrays.asList(dummyEventListener));
+    CatalogDispatcher catalogDispatcher = mockCatalogDispatcher();
+    this.dispatcher = new CatalogEventDispatcher(eventBus, catalogDispatcher);
+    CatalogDispatcher catalogExceptionDispatcher = mockExceptionCatalogDispatcher();
+    this.failureDispatcher = new CatalogEventDispatcher(eventBus, catalogExceptionDispatcher);
+  }
+
+  @Test
+  void testCreateCatalog() {
```

Comment:
I think you'd better rename the method to `testCreateCatalogEvent` here and below? Actually, you're testing the event, not the create operation, right?

Reply:
yes

Reply:
done